pvanalytics/features/daytime.py

```diff
@@ -229,3 +230,115 @@ def power_or_irradiance(series, outliers=None,
         correction_window
     )
     return ~night_corrected_edges
+
+
+def get_sunrise(daytime_mask, freq=None, data_alignment='L'):
+    """
+    Using the outputs of power_or_irradiance(), derive sunrise values for
+    each day in the associated time series.
+    Parameters
+    ----------
+    daytime_mask_series  : Series
+        Boolean series delineating night periods from day periods, where
+        day is True and night is False.
+    freq : str, optional
+        A pandas freqstr specifying the expected timestamp spacing for
+        the series. If None, the frequency will be inferred from the index.
+    data_alignment  : String, default 'L'
+        The data alignment of the series (left-aligned or right-aligned). Data
+        alignment affects the value selected as sunrise. Options are 'L' (left-
+        aligned), 'R' (right-aligned), or 'C' (center-aligned)
+    Returns
+    -------
+    Series
+        Series of daily sunrise times, based on the daytime_mask series.
+        This series has the same index as the passed daytime_mask series.
+    """
+    # Get the first day period for each day
+    sunrise_series = daytime_mask.index[daytime_mask].to_series().groupby(
+        daytime_mask[daytime_mask].index.date).transform('first').reindex(
+            daytime_mask.index)
```

Comment:
These are both really valid points that are also really tricky. For point #1, we could recommend running a data completeness check beforehand to throw out any gaps where there are large data holes, or I could just directly integrate that function in with a passable threshold for the amount of data required per day.

Point #2 is even trickier. I haven't dealt with this case in fleets as we try to localize the data as much as possible, but if you were going in blind with no idea what the timezone was, this could be problematic. We would need to add in logic that checks if the day period runs to the next day, and apply new logic to handle that.

Let me know your thoughts on the above.

Reply:
I think trying to solve these ourselves would be a can of worms.  I'd say let's document the assumptions/requirements and put the onus on the user to sort out any issues.  Something like this?

> This function assumes that each midnight-to-midnight period (according to the timezone of the input data) has one sunrise followed by one sunset. In cases where this is not satisfied (timezone of data is substantially different from the location's local time, locations near the poles, etc), or in the case of missing data, the estimated sunrise and sunset times may be invalid.

Reply:
@kandersolar added this to the comments in https://github.com/pvlib/pvanalytics/pull/187/commits/7fb4bf3f3da505a8be61f0376b8ce2935d4494ab. Makes my life easier to just include the caveat for the time being! Let me know if you think it'd be better suited in a 'Notes' section for the function.